exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/log_exporter/__init__.py

```diff
@@ -0,0 +1,188 @@
+    get_resource_data,
+    _translate_value,
+)
+from opentelemetry.proto.collector.logs.v1.logs_service_pb2 import (
+    ExportLogsServiceRequest,
+)
+from opentelemetry.proto.collector.logs.v1.logs_service_pb2_grpc import (
+    LogsServiceStub,
+)
+from opentelemetry.proto.common.v1.common_pb2 import InstrumentationLibrary
+from opentelemetry.proto.logs.v1.logs_pb2 import (
+    InstrumentationLibraryLogs,
+    ResourceLogs,
+    SeverityNumber,
+)
+from opentelemetry.proto.logs.v1.logs_pb2 import LogRecord as PB2LogRecord
+from opentelemetry.sdk.logs import LogRecord as SDKLogRecord
+from opentelemetry.sdk.logs import LogData
+from opentelemetry.sdk.logs.export import LogExporter, LogExportResult
+
+
+class OTLPLogExporter(
+    LogExporter,
+    OTLPExporterMixin[SDKLogRecord, ExportLogsServiceRequest, LogExportResult],
+):
+
+    _result = LogExportResult
+    _stub = LogsServiceStub
+
+    def __init__(
+        self,
+        endpoint: Optional[str] = None,
+        insecure: Optional[bool] = None,
+        credentials: Optional[ChannelCredentials] = None,
+        headers: Optional[Sequence] = None,
+        timeout: Optional[int] = None,
+        compression: Optional[Compression] = None,
+    ):
+        super().__init__(
+            **{
+                "endpoint": endpoint,
+                "insecure": insecure,
+                "credentials": credentials,
+                "headers": headers,
+                "timeout": timeout,
+                "compression": compression,
+            }
+        )
+
+    def _translate_name(self, log_data: LogData) -> None:
+        self._collector_log_kwargs["name"] = log_data.log_record.name
+
+    def _translate_time(self, log_data: LogData) -> None:
+        self._collector_log_kwargs[
+            "time_unix_nano"
+        ] = log_data.log_record.timestamp
+
+    def _translate_span_id(self, log_data: LogData) -> None:
+        self._collector_log_kwargs[
+            "span_id"
+        ] = log_data.log_record.span_id.to_bytes(8, "big")
+
+    def _translate_trace_id(self, log_data: LogData) -> None:
+        self._collector_log_kwargs[
+            "trace_id"
+        ] = log_data.log_record.trace_id.to_bytes(16, "big")
+
+    def _translate_trace_flags(self, log_data: LogData) -> None:
+        self._collector_log_kwargs["flags"] = int(
+            log_data.log_record.trace_flags
+        )
+
+    def _translate_body(self, log_data: LogData):
+        self._collector_log_kwargs["body"] = _translate_value(
+            log_data.log_record.body
+        )
+
+    def _translate_severity_text(self, log_data: LogData):
+        self._collector_log_kwargs[
+            "severity_text"
+        ] = log_data.log_record.severity_text
+
+    def _translate_attributes(self, log_data: LogData) -> None:
+        if log_data.log_record.attributes:
+            self._collector_log_kwargs["attributes"] = []
+            for key, value in log_data.log_record.attributes.items():
+                try:
```

Comment:
Why is this `try` necessary?

Reply:
This was needed because `_translate_key_values` raises an exception when it doesn't find a way to translate value.